compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -154,17 +154,19 @@ pub fn parse_asm_args<'a>(
         } else if p.eat_keyword(kw::Const) {
             let anon_const = p.parse_anon_const_expr()?;
             ast::InlineAsmOperand::Const { anon_const }
-        } else if !is_global_asm && p.eat_keyword(sym::sym) {
+        } else if p.eat_keyword(sym::sym) {
             let expr = p.parse_expr()?;
-            match expr.kind {
-                ast::ExprKind::Path(..) => {}
-                _ => {
-                    let err = diag
-                        .struct_span_err(expr.span, "argument to `sym` must be a path expression");
-                    return Err(err);
-                }
-            }
-            ast::InlineAsmOperand::Sym { expr }
+            let ast::ExprKind::Path(qself, path) = &expr.kind else {
+                let err = diag
+                    .struct_span_err(expr.span, "argument to `sym` must be a path expression");
```

Comment:
Wouldn't “expected a path” be more consistent with other messages that come this macro and the frontend in general? The fact that this is parsing a `sym` operand will usually be quite apparent from the code snippet around the span.

Additionally, I think there might be no tests exercising this branch, please add one if that's the case.

Reply:
I've updated the wording to "expected a path for argument to `sym`". This is already tested in the parse-error.rs asm ui tests.